### ISSUE TITLE
Add HttpApp.startServer(host, port, system)

### ISSUE THREAD

**Issue by ktoso** [akka/akka-http#1294]
Currently we require passing in the config and system if one wants to pass in system.
I just want to pass in the system, and have the default config.

Encountered this painfully in real code where I needed to start an HttpApp from a Java codebase, but the server was Scala. Creating the Scala settings object is a huge pain in Java due to the type system not managing well the apply return type of Akka Extensions..

Adding:

```
def startServer(host: String, port: Int, system: ActorSystem): Unit = {
    super.startServer(host, port, ServerSettings(system.settings.config), system)
  }
```

Will enable this use case, and is also just a missing overload I think

**Comment by ktoso** [comment]
FYI @jlprat

**Comment by jlprat** [comment]
I completely agree. I will take this one :) This overloaded crossed my mind as well and I thought I added an issue for it.

**Comment by ktoso** [comment]
Great, thanks!

**Comment by jlprat** [comment]
Closed via #1299
